Adding DI.External to runtime with its history

maryamariyan opened the pull request.

pakrym committed:
Add spec tests for external container implementations (dotnet/Extensions#629)



Commit migrated from https://github.com/dotnet/Extensions/commit/5b3645a13c66b49d24e9a655c25a51ac1f673219

Commit:
Merge branch 'release/2.2'


Commit migrated from https://github.com/dotnet/Extensions/commit/604018cb12e8b93e7423218bfd00efa330a75b28

natemcmaster committed:
Disable transitive project references in test projects (dotnet/Extensions#1834)



Commit migrated from https://github.com/dotnet/Extensions/commit/e7d5bea3bbd9c942c523e9068b17f11bb35f332b

davidfowl committed:
Added verification test for resolving singleton from scoped container. (dotnet/Extensions#2236)

* Added verification test for resolving singleton from scoped container.

* Skip failing test with the Unity container


Commit migrated from https://github.com/dotnet/Extensions/commit/732d143b501650bf4dda655d13b81549d60ae83f

Commit:
Support netcoreapp3.1 TFM (dotnet/Extensions#2336)

* Support netcoreapp3.1 TFM

* Unpin SDK for source build

* Update to preview1 branding


Commit migrated from https://github.com/dotnet/Extensions/commit/32cc8162ff38356b293611138d361f757a62e025

Commit:
Merge branch 'release/3.1'


Commit migrated from https://github.com/dotnet/Extensions/commit/785e2b0905844597546d4963e43851f000e4ee31

sharwell committed:
Normalize all file headers to the expected Apache 2.0 license


Commit migrated from https://github.com/dotnet/Extensions/commit/cec6e75f0ce3145c76acd4c8a8c14093696a03ba

sharwell committed:
Switch file headers to the MIT license


Commit migrated from https://github.com/dotnet/Extensions/commit/321a30c4d74b387db587d92f965391235fe33770

maryamariyan committed:
Merge branch 'di-external' of /Users/maryam/CodeHub/Extensions into di-external

maryamariyan committed:
Merge commit 'a6e9098ffc9adf987be1' into di-external

tarekgh commented:
Thanks @maryamariyan for brining in these tests

safern commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:
```diff
@@ -0,0 +1,39 @@
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\..\..\Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\ExceptionAssertions.cs">
```
Use `CommonTestPath` MSBuild property.

safern commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:
```diff
@@ -0,0 +1,39 @@
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
+    <EnableDefaultItems>true</EnableDefaultItems>
+    <SignAssembly>false</SignAssembly>
```
Why can't we sign this assembly? We use a Test key.

tarekgh replied:
These tests are using 3ed parties packages which including non-signed assemblies.

maryamariyan replied:
Added NoWarn so we could sign this as a test assembly

safern commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:
```diff
@@ -24,16 +22,12 @@
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
     <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.3.1.0"/>
```
This version is weird, is it really `4.3.1.0` or just `4.3.1`?

maryamariyan commented:
moved files slightly so that DI tests now contain 3 folders:

- DI.External.Tests (with its own csproj)
- DI.Specification.Tests (used in the other two)
- DI.Tests (with its own csproj)

maryamariyan commented:
Failures on net472:
```
System.IO.FileLoadException : Could not load file or assembly 'Lamar, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
   at Microsoft.Extensions.DependencyInjection.Specification.LamarDependencyInjectionSpecificationTests.CreateServiceProviderImpl(IServiceCollection serviceCollection)
   at Microsoft.Extensions.DependencyInjection.Specification.DependencyInjectionSpecificationTests.OuterServiceCanHaveOtherServicesInjected() in /_/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/DependencyInjectionSpecificationTests.cs:line 230

System.IO.FileLoadException : Could not load file or assembly 'DryIoc, Version=3.0.2.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
   at Microsoft.Extensions.DependencyInjection.Specification.DryIocDependencyInjectionSpecificationTests.CreateServiceProvider(IServiceCollection serviceCollection)
   at Microsoft.Extensions.DependencyInjection.Specification.DependencyInjectionSpecificationTests.ServicesRegisteredWithImplementationTypeCanBeResolved() in /_/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/DependencyInjectionSpecificationTests.cs:line 24
```

maryamariyan committed:
Enables DI.External
Moving around files in DI tests, correcting folder structure
Disable tests failing on mono and for netfx

maryamariyan committed:
Merge commit '6ccc72a9df472b2e1efd10d0c3493d580a9992ca' into di-external

maryamariyan commented:
The CI already went green. I just force pushed my changes now so that my commits-only were squashed together, keeping the commit history intact, doing a merge commit to submit the PR.
